url56b1dc76d3_wufunxiang_Libs_tgz-pJ8-org_msgpack_util_TemplatePrecompilerJ8/src/org/msgpack/template/builder/beans/IndexedPropertyDescriptor.java

```diff
@@ -0,0 +1,383 @@
+        } catch (SecurityException exception) {
+            // custom.beans.59=Security violation accessing indexed read method
+            throw new IntrospectionException(Messages.getString("custom.beans.59")); //$NON-NLS-1$
+        }
+        internalSetIndexedReadMethod(getter);
+    }
+
+    private void internalSetIndexedReadMethod(Method indexGetter)
+            throws IntrospectionException {
+        // Clearing the indexed read method.
+        if (indexGetter == null) {
+            if (indexedSetter == null) {
+                if (getPropertyType() != null) {
+                    // custom.beans.5A=Indexed method is not compatible with non indexed method
+                    throw new IntrospectionException(Messages
+                            .getString("custom.beans.5A")); //$NON-NLS-1$
+                }
+                indexedPropertyType = null;
+            }
+            this.indexedGetter = null;
+            return;
+        }
+        // Validate the indexed getter.
+        if ((indexGetter.getParameterTypes().length != 1)
+                || (indexGetter.getParameterTypes()[0] != Integer.TYPE)) {
+            // custom.beans.5B=Indexed read method must take a single int argument
+            throw new IntrospectionException(Messages.getString("custom.beans.5B")); //$NON-NLS-1$
+        }
+        Class<?> indexedReadType = indexGetter.getReturnType();
+        if (indexedReadType == Void.TYPE) {
+            // custom.beans.5B=Indexed read method must take a single int argument
+            throw new IntrospectionException(Messages.getString("custom.beans.5B")); //$NON-NLS-1$
+        } else if (indexedSetter != null
+                && indexGetter.getReturnType() != indexedSetter
+                        .getParameterTypes()[1]) {
+            // custom.beans.5A=Indexed read method is not compatible with indexed write method
+            throw new IntrospectionException(Messages.getString("custom.beans.5A")); //$NON-NLS-1$
+        }
+
+        // Set the indexed property type if not already set, confirm validity if
+        // it is.
+        if (this.indexedGetter == null) {
+            indexedPropertyType = indexedReadType;
+        } else {
+            if (indexedPropertyType != indexedReadType) {
+                // custom.beans.5A=Indexed read method is not compatible with indexed write method
+                throw new IntrospectionException(Messages.getString("custom.beans.5A")); //$NON-NLS-1$
+            }
+        }
+
+        // Set the indexed getter
+        this.indexedGetter = indexGetter;
+    }
+
+    private void setIndexedWriteMethod(Class<?> beanClass, String indexedSetterName)
+            throws IntrospectionException {
+        Method setter = null;
+        try {
+            setter = beanClass.getMethod(indexedSetterName, new Class[] {
+                    Integer.TYPE, getPropertyType().getComponentType() });
+        } catch (SecurityException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 309.